build.sbt

```
@@ -179,6 +180,16 @@ def runWebpack(file: File): Int = {
   }
 }
 
+npmCi := {
+  Changes.ifChanged(
+    target.value / "npm-tracking",
+    baseDirectory.value / "package.json",
```

```suggestion
    baseDirectory.value / "package-lock.json",
```

As you suggested, I think package-lock.json is a better candidate here, though the fact that you have `node_modules` tracked probably means it's fine

---

Agree. Probably don't need to track node_modules as well but I don't think I made the output optional in `Changes`. It doesn't seem to spend a lot of time checking for differences so assume SBT's using some smart filesystem stuff